main.nf

```diff
@@ -172,6 +171,25 @@ workflow {
     )
 }
 
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    LOCAL FUNCTIONS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+//
+// Get attribute from genome config file e.g. fasta
+//
+
+def getGenomeAttribute(attribute) {
+    if (params.genomes && params.genome && params.genomes.containsKey(params.genome)) {
+        if (params.genomes[ params.genome ].containsKey(attribute)) {
+            return params.genomes[ params.genome ][ attribute ]
+        }
+    }
+    return null
+}
```

Comment:
Hmmm...not sure about moving this here. Can we pass the `params` object to the function instead if it's a scoping issue?

Reply:
yeah, I opened this PR to play around with this.
I just reproduced the behavior from @ramprasadn PR's on raredisease, and then my plan was to try things out

Reply:
One of the approaches I tried to fix this issue involved passing `params` object to the function, and I don't think the problem is related to scoping. It appears that parameters are merged based on config priority and passed to workflows before they are even executed. This means that any modifications made to parameters within a workflow don't modify the original `params` object, and the changes are not visible to other workflows. 

This issue may be comparable to the limitation of not being able to set "NXF_*" environmental variables within the env scope.